lib/talkable/resources/offer.rb

```diff
@@ -5,5 +5,46 @@ def self.parse(result_hash)
       offer.claim_links ||= Hashie::Mash.new(result_hash[:claim_links])
       offer
     end
+
+    def advocate_share_iframe(options = {})
+      show_trigger = !options[:ignore_trigger]
+      tag = campaign_tags.first
+      iframe_options = default_iframe_options(tag).merge(options[:iframe] || {})
+      url = show_trigger ? Furi.update(show_url, query: {trigger_enabled: 1}) : show_url
+
+      snippets = []
+      if !options[:iframe] || !options[:iframe][:container]
+        snippets << render_container_snipet(iframe_options[:container])
+      end
+      snippets << render_share_snipet({
+        url: url,
+        iframe: iframe_options,
+      })
+
+      snippets.join("\n")
+    end
+
+    protected
+
+    def default_iframe_options(tag = nil)
+      tag ||= SecureRandom.hex(3)
+      {
+        container: "talkable-offer-#{tag}",
+        width: '100%',
+      }
+    end
+
+    def render_container_snipet(name)
+      "<div id='#{name}'></div>"
```

Comment:
You may need to copy the `content_tag` from ActionView. It maybe easier than inventiting it each time.